src/EditorFeatures/Core/Implementation/Suggestions/SuggestedActionsSource.cs

```diff
@@ -891,33 +891,32 @@ private void OnSuggestedActionsChanged(Workspace currentWorkspace, DocumentId cu
                 var provider = _owner;
                 using (var asyncToken = _owner.OperationListener.BeginAsyncOperation(nameof(GetSuggestedActionCategoriesAsync)))
                 {
-                    var selection = await GetSpanAsync(range).ConfigureAwait(false);
-                    if (selection != null)
+                    var document = range.Snapshot.GetOpenDocumentInCurrentContextWithChanges();
+                    using (var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken))
                     {
-                        var document = range.Snapshot.GetOpenDocumentInCurrentContextWithChanges();
-                        using (var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken))
+                        var linkedToken = linkedTokenSource.Token;
+
+                        var errorTask = Task.Run(
+                            () => GetFixLevelAsync(provider, document, range, linkedToken), linkedToken);
+
+                        var selection = await GetSpanAsync(range).ConfigureAwait(false);
+                        Task<string> refactoringTask = Task.FromResult((string)null);
+                        if (selection != null && requestedActionCategories.Contains(PredefinedSuggestedActionCategoryNames.Refactoring))
```

Comment:
You're correct--it only respects the presence of the refactoring category. Looking at some of the PredefinedSuggestedActionCategoryNames, there are categories like "StyleFix" that currently don't map onto our severity model. For that reason, I'd prefer to not attempt to fully plumb through all the remaining category names for now (given that respecting refactorings fixes the bug :)).